Scripts/Editor/NodeEditorGUI.cs

```diff
@@ -117,39 +117,80 @@ public partial class NodeEditorWindow {
                         if (i < length - 2) {
-                            Vector2 ab = (windowPoints[i + 1] - windowPoints[i]).normalized;
-                            Vector2 cb = (windowPoints[i + 1] - windowPoints[i + 2]).normalized;
-                            Vector2 ac = (windowPoints[i + 2] - windowPoints[i]).normalized;
+							Vector2 point_c = windowPoints[i + 2];
+                            Vector2 ab = (point_b - point_a).normalized;
+                            Vector2 cb = (point_b - point_c).normalized;
+                            Vector2 ac = (point_c - point_a).normalized;
                             Vector2 p = (ab + cb) * 0.5f;
-                            float tangentLength = (Vector2.Distance(windowPoints[i], windowPoints[i + 1]) + Vector2.Distance(windowPoints[i + 1], windowPoints[i + 2])) * 0.005f * zoom;
-                            float side = ((ac.x * (windowPoints[i + 1].y - windowPoints[i].y)) - (ac.y * (windowPoints[i + 1].x - windowPoints[i].x)));
+                            float tangentLength = (dist_ab + Vector2.Distance(point_b, point_c)) * 0.005f * zoom;
+                            float side = ((ac.x * (point_b.y - point_a.y)) - (ac.y * (point_b.x - point_a.x)));
 
                             p = new Vector2(-p.y, p.x) * Mathf.Sign(side) * tangentLength;
                             inputTangent = p;
                         }
                         else {
-                            inputTangent = Vector2.left * Vector2.Distance(windowPoints[i], windowPoints[i + 1]) * 0.01f * zoom;
+                            inputTangent = Vector2.left * dist_ab * 0.01f * zoom;
                         }
 
-                        Handles.DrawBezier(windowPoints[i], windowPoints[i + 1], windowPoints[i] + ((outputTangent * 50) / zoom), windowPoints[i + 1] + ((inputTangent * 50) / zoom), col, null, 4);
+						// Calculates the tangents for the bezier's curves.
+						Vector2 tangent_a = point_a + outputTangent * 50 / zoom;
+						Vector2 tangent_b = point_b + inputTangent * 50 / zoom;
+						// Hover effect.
+						int bezier_width = 4;
+						int division = Mathf.RoundToInt(.1f * dist_ab) + 3;
+						Vector3[] points = Handles.MakeBezierPoints(point_a, point_b, tangent_a, tangent_b, division);
+						if (!disableHover) {
+							for (int j = 0; j < points.Length; j++) {
+								Vector3 current = points[j];
+								bool next_has_mouse = false;
+								if (j + 1 < points.Length) {
+									Vector3 next = points[j + 1];
+									next_has_mouse = ContainsMouse(current, next);
+								}
+								if (next_has_mouse) {
+									bezier_width += 2;
+									break;
+								}
+							}
+						}
+						// Coloring and bezier drawing.
+						for (int j = 0; j < points.Length - 1; j++)	{
+							Handles.color = grad.Evaluate((j + 1f) / (points.Length));
+							Handles.DrawAAPolyLine(bezier_width / zoom, points[j], points[j + 1]);
+						}

```

Comment:
I appreciate tab indentation but please convert into 4 spaces to remain consistent with the rest of the file